File: 이승연/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
import java.util.*;

class Solution {
    public int solution(String[] babbling) {
        String[] words = {"aya", "ye", "woo", "ma"};
        int answer = 0;
        
        for (int i = 0; i < babbling.length; i++) {
            int strCount = 0;
            for (String word : words) {
                if (babbling[i].contains(word)) {
                    strCount += word.length();
                }
            }
            
            if (babbling[i].length() == strCount) {
                answer += 1;
            }
        }
        
        return answer;
    }
}

File: 이승연/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.js
function solution(babbling) {
    const words = ["aya", "ye", "woo", "ma"];
    let answer = 0;
    
    for (let i = 0; i < babbling.length; i++) {
        let count = 0;
        let str = babbling[i];
        let visited = new Array(4).fill(0);
        
        while(count < 4) {
            for (let j = 0; j < words.length; j++) {
                if (visited[j] === 0 && str.indexOf(words[j]) === 0) {
                    str = str.replace(words[j], "");
                    visited[j] = 1;
                }
            }
            count++;
        }
        
        if (str.length === 0) {
            answer++;
        }
    }
    
    return answer;
}